dbt/include/databricks/macros/materializations/incremental/strategies.sql

```diff
@@ -51,8 +51,17 @@
 {% macro get_insert_into_sql(source_relation, target_relation) %}
 
     {%- set dest_columns = adapter.get_columns_in_relation(target_relation) -%}
-    {%- set dest_cols_csv = dest_columns | map(attribute='quoted') | join(', ') -%}
-    insert into table {{ target_relation }}
+    {%- set source_columns = adapter.get_columns_in_relation(source_relation) -%}
+    {%- set common_columns = [] -%}
+    {%- for dest_col in dest_columns -%}
+        {%- for source_col in source_columns -%}
+            {%- if dest_col.name == source_col.name -%}
+                {%- if common_columns.append(dest_col) -%}{%- endif -%}
```

Comment:
This seems like it will not work in the case where columns are added in the new code.  Also, I'm not sure this is any more efficient, but stylistically it might be better to do a single loop and check for presence in the second list:

```
{%- set source_columns = adapter.get_columns_in_relation(source_relation) | map(attribute="name") -%}
{%- for dest_col in dest_columns -%}
   {% if dest_col.name in source_columns %}
...
```

Reply:
Thank you for your review year I will refactor it!

>This seems like it will not work in the case where columns are added in the new code

I think default case is ignore and existing code is also using destination columns. So I thought Adding new columns might not be problem. How do you think?

Reply:
I may not be familiar enough with the surrounding code, let me look a little deeper.

Reply:
Ok, I understand now how this works, I think this is fine.

Reply:
Let me try explaining it to verify:

This statement is saying that in the target relation (which is the existing table), we are going to insert into it all the new incremental data that matches from the tmp source table; `insert into` will provide default values for any columns not named.  The issue you are solving is that previously we attempted to select columns from source that were only guaranteed to be in target.

What I'm confused by is why no one has raised this issue on dbt-spark :)

Reply:
@benc-db  Thank you for testing and investigation. 
I see. Yea I agree and wander why no one has raised this issue till this time. 

I refactor this as you suggested. Could you please check this?
https://github.com/databricks/dbt-databricks/pull/580/commits/b33764ca2569a9d0f87d04a803486b888dab648b

Reply:
Hi @case-k-git , This single loop iteration is breaking functionality and require an immediate fix.
This will only work when source and destination have columns in same order, as a result we forcefully required to perform `--full-refresh`. For an instance: 
<img width="1267" alt="image" src="https://github.com/databricks/dbt-databricks/assets/24986964/e7af9eed-85cd-4aaf-b5e6-d1e65ff4a7d6">

DBT's internal Temp view (source columns: [A, B, C, ... N]) and Destination table (target columns arranged in random fashion due to some process: [A, B, C, D, F, E, .. N])  have actually 14 columns in common, but due to this single iteration change, it compares sequentially and returns only 4 & as a result, inserts 10 blank columns at the time of final execution. Looking forward for the immediate resolutions and test cases as well. Thank you.
CC: @benc-db

Reply:
@nimeshpatni  Thank you! I see let me check!

Reply:
I confirm the error by changing column order . Thank you
https://github.com/databricks/dbt-databricks/issues/594#issuecomment-1978019472